### PR TITLE
Adding Profiling for script engine timer callback

### PR DESCRIPTION
**Pull request by samcake**
This pr add a profiling of the time interval call back to script functions in the scripting engine.

This pr also fixes tool scripts that got broken recently which we use to monitor the rendering engine.
see the comment in PerfPlot.qml

Finally, this pr introduce a new way to access a job config through getConfig(path) allowing to express  a path (or partial path) to a job in a task graph.
see comment in render/src/task/Config.h

## TEST PLAN
Normal usage shouldn't have changed, smoke test should show no problem.

Any of the scripts from Developer/utilities/render that rely on the PlotPerf.qml should work just fine
the plotting should just work, these are:
 renderStatsGPU.js, renderStats.js, textureMonitor.js, renderRates.js, lightClustering.js
The other scripts there are also fixed somewhat by this pr but these are  mostly specific tools  to debug the engine, let s not worry about that for QA here.

### CODE REVIEW

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 1]
```diff
@@ -1283,6 +1284,7 @@ void ScriptEngine::timerFired() {
 
     // call the associated JS function, if it exists
     if (timerData.function.isValid()) {
+        PROFILE_RANGE(script, __FUNCTION__);
```
Comment (samcake) [inline libraries/script-engine/src/ScriptEngine.cpp]: adding the profiling of the timer call back to scripts

File: scripts/developer/utilities/lib/plotperf/PlotPerf.qml [hunk 2]
```diff
@@ -49,11 +49,15 @@ Item {
 
     property var valueMax : 1
 
-    property var _values : new Array()
+    property var _values
```
Comment (samcake) [inline scripts/developer/utilities/lib/plotperf/PlotPerf.qml]: this was the problem
within the last few days, the behavior of qml changed and that _values initialization was happening twice.
once before calling Component.onCompleted
and then apparently once more after.... which erased what was created in the "createValues()' call.

**Comment by CainFoool** [comment]
Testing once ready.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/942/HighFidelity-Beta-PR10765-09529f6592b3e58228ede39b666ae47a40068fef.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/942/HighFidelity-Beta-PR10765-09529f6592b3e58228ede39b666ae47a40068fef.exe

**Comment by CainFoool** [comment]
# Test Passed 

Smoke test proved that everything seems to be working okay, as expected. 

Scripts mentioned in OP also appear to be functioning and rendering okay. (See screenshot) 
image

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Pull request merged**